src/net/unix/stream.rs

```diff
@@ -96,24 +111,31 @@ impl UnixStream {
     /// This function will return the first error that [`write`] returns.
     ///
     /// [`write`]: Self::write
-    pub async fn write_all<T: IoBuf>(&self, mut buf: T) -> crate::BufResult<(), T> {
+    pub async fn write_all<T: IoBuf>(
+        &self,
+        buf: impl Into<Slice<T>>,
+    ) -> crate::BufResult<(), Slice<T>> {
+        self.write_all_slice(buf.into()).await
+    }
+
+    // The implementation of write_all, factored out to reduce code bloat.
+    async fn write_all_slice<T: IoBuf>(&self, mut buf: Slice<T>) -> crate::BufResult<(), Slice<T>> {
         // This function is copied from the TcpStream impl.
-        let mut n = 0;
-        while n < buf.bytes_init() {
-            let res = self.write(buf.slice(n..)).await;
+        let (in_begin, in_end) = (buf.begin(), buf.end());
```

Comment:
Nice catch, getting tcp and unix in sync again. Another reason it will be nice to have one trait that covers both write_all in the future.